File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    arr = [i if i >= 2 else 0 for i in range(n + 1)]
    for i in range(2, int(n**0.5) + 1):
        for j in range(i*2, n + 1, i):
            if arr[j] == 0: continue;
            if arr[j] % i == 0: arr[j] = 0
    return len([i for i in arr if i != 0])

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    const arr = Array(n + 1).fill(0).map((_, i) => i >= 2 ? i : 0)
    for(let i = 2; i <= Math.sqrt(n); i++){
        for(let j = i * 2; j < arr.length; j += i){
            if(arr[j] === 0) continue;
            if(arr[j] % i === 0) arr[j] = 0
        }
    }
    return arr.filter(el => el !== 0).length
}